### PR TITLE
Allow to filter products via API only by enabled taxon

### PR DESCRIPTION
**Pull request by jakubtobiasz**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13
| Bug fix?        | kinda?
| New feature?    | kinda
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: features/product/viewing_products/accessing_disabled_taxon.feature [hunk 1]
```diff
@@ -7,9 +7,17 @@ Feature: Accessing a disabled taxon
...
     Scenario: Accessing a disabled taxon
         Given the store has "T-Shirts" taxonomy
         And the "T-Shirts" taxon is disabled
         When I try to browse products from taxon "T-Shirts"
         Then I should be informed that the taxon does not exist
+
+    @api @no-ui
+    Scenario: Filtering products by taxon available only for enabled taxon
+        Given the store has "Food" taxonomy
+        Given the store has a product "Baguette" priced at "$2.00" belonging to the "Food" taxon
```
Comment (GSadee) [inline features/product/viewing_products/accessing_disabled_taxon.feature:20]:
```suggestion
        And the store has a product "Baguette" priced at "$2.00" belonging to the "Food" taxon
```

File: src/Sylius/Behat/Context/Api/Shop/ProductContext.php [hunk 2]
```diff
@@ -648,6 +648,14 @@ public function iShouldBeInformedThatTheProductDoesNotExist(): void
         Assert::same($this->client->getLastResponse()->getStatusCode(), Response::HTTP_NOT_FOUND);
     }
 
+    /**
+     * @Then /^I should be informed that the taxon does not exist$/
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/ProductContext.php:652]:
```suggestion
     * @Then I should be informed that the taxon does not exist
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Jacob! :tada:

File: src/Sylius/Behat/Context/Api/Shop/ProductContext.php [hunk 3]
```diff
@@ -648,6 +648,14 @@ public function iShouldBeInformedThatTheProductDoesNotExist(): void
...
+     */
+    public function iShouldBeInformedThatTheTaxonDoesNotExist(): void
+    {
+        Assert::same($this->client->getLastResponse()->getStatusCode(), Response::HTTP_NOT_FOUND);
+    }
+
```
Comment (Rafikooo) [inline src/Sylius/Behat/Context/Api/Shop/ProductContext.php:658]: As far as I can see correctly this one isn't used

**Pull request merged**